app/controllers/api/v8/exercises_controller.rb

```diff
@@ -72,13 +72,11 @@ def get_by_course
       respond_not_found('Course not found!')
     else
       exercises = Exercise.where(course_id: course.id)
-
       visible = exercises.select { |ex| ex.visible_to?(current_user) }
-
       presentable = visible.map do |ex|
         {
             id: ex.id,
-            available_points: Exercise.find_by(id: ex.id).available_points,
+            available_points: ex.available_points,
```

Comment:
This helps, but not by much...
The `available_points` likely still causes a sql query, try with `includes` http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations